packages/ui-test-queries/src/utils/bindElementToEvents.ts

```diff
@@ -26,42 +26,65 @@ import keycode from 'keycode'
 
 import { isElement } from './isElement'
 import { find } from './queries'
-import { GenericFunction } from './bindElementToMethods'
-import { FireEvent, FireEventMethod } from './events'
+import { eventMap, FireEvent, FireEventInit, FireEventMethod } from './events'
+
+export type EventMapTypes = {
+  // every fireXYEvent looks the same except keyboard and blur ones.
+  [K in Extract<
+    keyof Omit<typeof eventMap, 'keyDown' | 'keyPress' | 'keyUp' | 'blur'>,
+    string
+  >]: (
+    init?: FireEventInit,
+    options?: Record<string, unknown>
+  ) => Promise<Event>
+} &
+  {
+    // blur event triggers handled by fireBlurEvent are a bit different
+    [K in 'blur']: (
+      init?: FireEventInit,
+      options?: Record<string, unknown>
+    ) => Promise<Event | void>
+  } &
+  {
+    // keyboard event triggers handled by fireKeyboardEvent are a bit different
+    [K in 'keyDown' | 'keyPress' | 'keyUp']: (
+      whichKey?: string | number,
+      init?: FireEventInit,
+      options?: Record<string, unknown>
+    ) => Promise<Event>
+  } // TODO refactor this to make it more uniform
```

Comment:
This is a horror looking type, but I did not want to introduce breaking changes. We should refactor this code to be nicer and more TS friendly, it has tons of unnecessary abstractions.